src/utils/deepAssign.js

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import isObject from "./isObject";
+
+const deepAssignObject = (target, source) => {
+  Object.keys(source).forEach(key => {
+    if (isObject(target[key]) && isObject(source[key])) {
+      deepAssignObject(target[key], source[key]);
+
+      return;
+    }
+
+    target[key] = source[key];
+  });
+};
+
+/**
+ * Recursively copy the values of all enumerable own properties from a source item to a target item if the both items are objects
+ * @private
+ * @param {Object} target - a target object
+ * @param {Object} source - a source object
+ * @example
+ * deepAssign({ a: 'a', b: 'b' }, { b: 'B', c: 'c' });
+ * // { a: 'a', b: 'B', c: 'c' }
+ */
+export default (target, ...sources) => {
+  if (!isObject(target)) {
+    return {};
+  }
+
+  const { length } = sources;
```

Comment:
This really isn't a big deal in any way, but caching length _generally_ doesn't help with performance anymore and can often hinder it. Here's a little on the issue: https://mrale.ph/blog/2014/12/24/array-length-caching.html In any case, it will make little difference either way.

Reply:
@Aaronius it's just a habit, not necessarily for performance reasons.

Reply:
After some code changes I replaced it with `forEach()`